src/mlsquare/architectures/sklearn.py

```diff
@@ -85,6 +87,84 @@ def update_params(self, params):
     def adapter(self):
         return self._adapter
 
+class DimensionalityReductionModel:
```

Comment:
class MatrixDecomposition(BaseModel):
and implement all methods in BaseModel with pass. Since BaseModel required create_model which is not required MatrixDecompositions, and add fit_transform as you defined here.

Reply:
So sir, you're suggesting i should define a new BaseModel class with just fit_transform method as abstract method, like the way it is done in class DimensionalityReductionModel but just with renaming it to MatrixDecomposition ?

Reply:
see comment below.